Sources/MapboxCoreNavigation/NavigationRouter.swift

```diff
@@ -0,0 +1,179 @@
+import Foundation
+import MapboxDirections
+import MapboxNavigationNative
+
+public class NavigationRouter {
+    
+    public typealias RequestId = UInt64
+    
+    public struct RoutingRequest {
+        public let id: RequestId
+        
+        // Intended retain cycle to prevent deallocating. `RoutingRequest` will be deleted once request completes.
+        let router: NavigationRouter
+        
+        public func cancel() {
+            router.finish(request: id)
+        }
+    }
+    
+    public enum RouterSource {
+        case online
+        case offline
+        case hybrid
+        
+        var nativeSource: RouterType {
+            switch self {
+            case .online:
+                return .online
+            case .offline:
+                return .onboard
+            case .hybrid:
+                return .hybrid
+            }
+        }
+    }
+    
+    public private(set) var activeRequests: [RequestId : RoutingRequest] = [:]
+    public private(set) var source: RouterSource
```

Comment:
Is RouterSource modified anywhere?

Reply:
Right, It should be a constant as well as other properties below.